### PR TITLE
Implemented public recipe

### PR DESCRIPTION
**Pull request by muneebulrehman**
# Implemented public recipe
- Made public recipe page as root path
- Displayed the publicly available recipes
- Can access the publicly available recipe.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
         <div class="border-b border-white/25 p-4 text-4xl font-thin">Railicious</div>
         <nav class="flex grow flex-col justify-between gap-4">
           <ul class="flex flex-col items-stretch justify-between">
-            <li><a href="#" class="block px-4 py-2 hover:bg-white/10">Public Recipes</a></li>
+            <li><%= link_to "Public Recipes", root_path, class:"block px-4 py-2 hover:bg-white/10" %></li>
```
Comment (ahangarha) [inline app/views/layouts/application.html.erb:18]: It is true that we show same page if we visit root, but we should have a dedicate path for it. Better to link it to public_recipes page. It is one of the requirements to have such path.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ def index
   end
 
   def show
-    if current_user != Recipe.find(params[:id]).user
+    if current_user != Recipe.find(params[:id]).user && !Recipe.find(params[:id]).public
       flash[:notice] = 'You are not authorized to access this page'
       redirect_to recipes_path
     end
```
Comment (ahangarha) [inline app/controllers/recipes_controller.rb:12]: ## Change Required
This doesn't work. I cannot see the page if I am not logged in.

File: app/models/recipe.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ class Recipe < ApplicationRecord
 
   validates :name, presence: true
 
-  def public_recipes
-    @recipes = Recipe.where(public: true)
+  def self.public_recipes
```
Comment (ahangarha) [inline app/models/recipe.rb:8]: Do we need to mention self here?

**Review by ahangarha — CHANGES_REQUESTED** [review]
Good work so far. But not complete yet.

# CHANGES REQUIRED

Please check the inline comments. Key issues are as following:
- We cannot see public recipes list without being logged in
- We cannot see public recipes detail without being logged in
- There is not route for `domain/public_recipes`. Consider changing public to public_recipes.
- Link in the navigation bar to the public_recipes leads to root path. Consider changing it to the public_recipes path.

File: app/models/recipe.rb (hunk 3 above)
Comment (muneebulrehman) [reply app/models/recipe.rb:8]: Yes, it is needed.

**Pull request merged**